### PR TITLE
RM - Adoption Survey

### PR DESCRIPTION
**Pull request by Diama1**
#997

### CODE REVIEW

File: client/src/elm/Backend/ResilienceSurvey/Model.elm [hunk 1]
```diff
@@ -29,6 +30,12 @@ type ResilienceSurveyQuestion
     | ResilienceSurveyQuestion8
     | ResilienceSurveyQuestion9
     | ResilienceSurveyQuestion10
+    | ResilienceSurveyQuestion11
```
Comment (anvmn) [inline client/src/elm/Backend/ResilienceSurvey/Model.elm:33]: Currently, we support up to 10 questions per survey.
How many questions are there at adoption survey? 16?
If not, drop unneeded options.

File: client/src/elm/Backend/ResilienceSurvey/Model.elm [hunk 2]
```diff
@@ -37,6 +44,11 @@ type ResilienceSurveyQuestionOption
     | ResilienceSurveyQuestionOption2
     | ResilienceSurveyQuestionOption3
     | ResilienceSurveyQuestionOption4
+    | ResilienceSurveyQuestionOption5
```
Comment (anvmn) [inline client/src/elm/Backend/ResilienceSurvey/Model.elm]: Currently, we support up to 4  answers for a question.
What's the maximal number of answers for question at adoption survey? 9?
If not, drop unneeded options.

**Review by anvmn — CHANGES_REQUESTED** [review]
@Diama1 

We should not introduce stuff we don't use.

File: client/src/elm/Backend/ResilienceSurvey/Model.elm (hunk 1 above)
Comment (Diama1) [reply client/src/elm/Backend/ResilienceSurvey/Model.elm:33]: they are 12

File: client/src/elm/Backend/ResilienceSurvey/Model.elm (hunk 2 above)
Comment (Diama1) [reply client/src/elm/Backend/ResilienceSurvey/Model.elm]: they are 5 and they are different from the 4 we had for the previous survey

Comment (anvmn) [reply client/src/elm/Backend/ResilienceSurvey/Model.elm]: How are they different? If question has up to 5 answers, we should have 5 options.

File: client/src/elm/Backend/ResilienceSurvey/Model.elm (hunk 1 above)
Comment (anvmn) [reply client/src/elm/Backend/ResilienceSurvey/Model.elm:33]: Then, there should be `ResilienceSurveyQuestion1 - ResilienceSurveyQuestion12`

Comment (Diama1) [reply client/src/elm/Backend/ResilienceSurvey/Model.elm:33]: I meant we have 4 of the first survey and 12 of the adoption survey

File: client/src/elm/Backend/ResilienceSurvey/Model.elm (hunk 2 above)
Comment (Diama1) [reply client/src/elm/Backend/ResilienceSurvey/Model.elm]: I get it

**Comment by Diama1** [comment]
survey

rmSurvey

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 3]
```diff
@@ -66,46 +66,64 @@ view language currentTime nurseId nurse db model =
...
-                                    viewQuarterlySurvey language currentDate nurseId model.surveyForm
-
-                                else if runAdoptionSurvey then
-                                    viewAdoptionSurvey language currentDate nurseId model.surveyForm
-
-                                else
-                                    messagingCenterView
-                            )
-                        |> Maybe.withDefault messagingCenterView
+                    if surveyCount == 0 then
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: You're counting here both ResilienceSurveyAdoption and ResilienceSurveyQuarterly.
This is wrong.

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 4]
```diff
@@ -66,46 +66,64 @@ view language currentTime nurseId nurse db model =
...
+                        if runQuarterlySurvey then
+                            viewQuarterlySurvey language currentDate nurseId model.surveyForm
+
+                        else
+                            viewAdoptionSurvey language currentDate nurseId model.surveyForm
+
+                    else if surveyCount == 3 then
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: You're counting here both ResilienceSurveyAdoption and ResilienceSurveyQuarterly.
This is wrong.

**Review by anvmn — CHANGES_REQUESTED** [review]
Revert to original code. All changes need to be within:

Selection_558

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 5]
```diff
@@ -70,36 +66,64 @@ view language currentTime nurseId nurse db model =
...
+                            let
+                                filterCondition survey =
+                                    let
+                                        diffLastSurveyCurrent =
+                                            Date.diff Months survey.dateMeasured currentDate
+
+                                        diffLastSurveyProgramStart =
+                                            Date.diff Months survey.dateMeasured programStartDate
+                                    in
+                                    if surveyCount surveyType == 0 then
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: Order of conditions is wrong

**Review by anvmn — CHANGES_REQUESTED** [review]

File: client/src/elm/Pages/MessagingCenter/View.elm (hunk 5 above)
Comment (anvmn) [reply client/src/elm/Pages/MessagingCenter/View.elm]: @Diama1 still wrong

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 6]
```diff
@@ -70,36 +66,64 @@ view language currentTime nurseId nurse db model =
...
+                        surveys =
+                            Dict.get nurseId db.resilienceSurveysByNurse
+                                |> Maybe.andThen RemoteData.toMaybe
+                                |> Maybe.map Dict.values
+                                |> Maybe.withDefault []
+
+                        surveysSorted =
+                            List.sortWith (sortByDateDesc .dateMeasured) surveys
+
+                        surveyCount surveyType =
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: This is not necessary.
Filtering should be done once, inside ` runSurvey surveyType`.

**Review by anvmn — CHANGES_REQUESTED** [review]

**Review by anvmn — CHANGES_REQUESTED** [review]
@Diama1 

I don't see this comment addressed:
Selection_561

**Comment by Diama1** [comment]
> @Diama1
> 
> I don't see this comment addressed: Selection_561

I meant we have 4 from quarterly survey and 12 of Adoption and that makes up 16 questions

**Review by anvmn — APPROVED** [review]

**Pull request merged**